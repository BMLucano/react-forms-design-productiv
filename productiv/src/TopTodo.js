import React from "react";

import Todo from "./Todo";

/** Shows the top todo.
 *
 * Props:
 * - todos: [{id, title, description, priority},...]
 *

 * TodoApp -> TopTodo
 */

function TopTodo({ todos }) {
  // lowest-priority # is the highest priority
  let top = todos.reduce(
    (acc, cur) => cur.priority < acc.priority ? cur : acc, todos[0]);

    //TODO: keep return on the same line if it's short
  return (
    <Todo todo={top} />
  );
}

export default TopTodo;