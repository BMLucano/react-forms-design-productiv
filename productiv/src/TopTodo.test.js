import React from "react";
import { render } from "@testing-library/react";
import TopTodo from './TopTodo'
import Todo from './Todo'


const testTodos = [{
  id: 1,
  title:"test",
  description:"test description",
  priority: 1},
  {
    id: 2,
    title:"test2",
    description:"test description2",
    priority: 2}
  ];

  let renderedTodo;

beforeEach(function(){
  renderedTodo = render(<TopTodo todos={testTodos} />)
  // debug(renderedTodo)
  // console.log(renderedTodo )
})

it("renders without crashing", function(){
  render(<TopTodo todos={testTodos} />)
});

it("matches snapshot", function(){
  expect(renderedTodo.asFragment()).toMatchSnapshot()
});

it("displays highest priority todo", function(){
  expect(renderedTodo.getByText("test")).toBeInTheDocument();
  expect(renderedTodo.getByText("test description")).toBeInTheDocument();
  expect(renderedTodo.getByText("(priority: 1)")).toBeInTheDocument();

})

it("does not display other todos", function(){
  expect(renderedTodo.queryByText("test2")).not.toBeInTheDocument();
  expect(renderedTodo.queryByText("test description2")).not.toBeInTheDocument();
  expect(renderedTodo.queryByText("(priority: 2)")).not.toBeInTheDocument();
});
