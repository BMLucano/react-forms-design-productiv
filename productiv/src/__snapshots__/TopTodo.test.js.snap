// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <div>
      <b>
        test
      </b>
       
      <small>
        (priority: 1)
      </small>
    </div>
    <div>
      <small>
        test description
      </small>
    </div>
  </div>
</DocumentFragment>
`;
